tools/walletextension/rpcapi/blockchain_api.go

```diff
@@ -177,7 +182,24 @@ func (api *BlockChainAPI) GetStorageAt(ctx context.Context, address gethcommon.A
 		}
 		return serialised, nil
 	default: // address was not a recognised custom query method address
-		return nil, fmt.Errorf("eth_getStorageAt is not supported on TEN")
+		if !api.storageWhitelist.AllowedStorageSlots[params] {
+			return nil, fmt.Errorf("eth_getStorageAt is not supported on TEN")
+		}
+
+		resp, err := ExecAuthRPC[any](ctx, api.we, &ExecCfg{tryUntilAuthorised: true}, "eth_getStorageAt", address.Hex(), params, nil)
+		if err != nil {
+			return nil, fmt.Errorf("unable to execute custom query: %w", err)
+		}
+		if resp == nil {
+			return nil, nil
+		}
+
+		respHex, ok := (*resp).(string)
```

Comment:
the response should be bytes, to avoid the conversion

Reply:
Having the response as bytes makes geth automatically encode it in base64 and the conversion remains. The pattern seems to be hexencoding bytes.